onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -49,13 +51,30 @@ std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_DML(ID
   return std::make_shared<onnxruntime::DMLProviderFactory>(dml_device, cmd_queue);
 }
 
+bool IsSoftwareAdapter(IDXGIAdapter1* adapter)
+{
+    DXGI_ADAPTER_DESC1 pDesc;
```

Comment:
>pDesc [](start = 23, length = 5)

`p`? Is this really a pointer to anything (and even if it is, didn't we kill Hungarian)?

Propose `adapterDesc`.